scripts/nbconverted/explore_relationship_AandB.py

```diff
@@ -0,0 +1,585 @@
+
+
+# Read dataframe with gene expression transformed
+predict_file = os.path.join(base_dir, "output", analysis_name, "predicted_gene_exp.txt")
+predict_gene_exp = pd.read_table(predict_file, header=0, index_col=0)
+
+print(predict_gene_exp.shape)
+predict_gene_exp.head()
+
+
+# In[12]:
+
+
+# Get the means of B genes
+
+# Convert dataframe with gene ids to list
+geneSetB_ls = [l[0] for l in geneSetB.values.tolist()]
+
+geneSetB_exp = predict_gene_exp[geneSetB_ls]
+
+# Get the mean for each sample
+geneSetB_mean = geneSetB_exp.mean(axis=1)
+geneSetB_mean.head()
+
+
+# In[13]:
+
+
+# Join original expression of A and mean(transformed expression of B)
+original_A_exp = test_samples[rep_gene_A]
+predict_B_mean_exp = geneSetB_mean
+
+A_and_B_df = pd.merge(original_A_exp.to_frame('gene A untransformed'),
+                      predict_B_mean_exp.to_frame('mean gene B transformed'),
+                      left_index=True, right_index=True)
+A_and_B_df.head()
+
+
+# In[14]:
+
+
+# Plot
+sns.regplot(x='gene A untransformed',
+            y='mean gene B transformed',
+           data = A_and_B_df)
+
+
+# ## What is the offset capturing?
+# 
+# How are the values of the latent features changing as A moves across threshold? i.e. What is the difference in the encodings when gene A expression is above vs below the threshold?
+# 
+# This is the shift we should be capturing in our offset vector
+# 
+# Currently we are using the extremes of gene A to capture the “essence of A” and what it means to “turn on”
+
+# In[15]:
+
+
+# Read VAE space offset
+offset_vae_space = pd.read_table(offset_vae_file, header=0, index_col=0)
+offset_vae_space
+
+
+# In[16]:
+
+
+# Read dataframe with gene expression transformed
+predict_encoded_file = os.path.join(base_dir, "output", analysis_name, "predicted_encoded_gene_exp.txt")
+predict_encoded_gene_exp = pd.read_table(predict_encoded_file, header=0, index_col=0)
+
+print(predict_encoded_gene_exp.shape)
+predict_encoded_gene_exp.head()
+
+
+# In[17]:
+
+
+data_encoded = pd.concat([offset_vae_space, predict_encoded_gene_exp], axis=0)
+data_encoded.head()
+
+
+# In[18]:
+
+
+# Add group labels per sample (<0.5, 0.5, 0.5>)
+samples = data_encoded.index
+
+data_encoded_labeled = data_encoded.assign(
+    threshold_group=(
+        list( 
+            map(
+                lambda x: 'offset' if x== 0 
+                else 'less' if test_samples_sorted.loc[x,rep_gene_A]<0.49
+                else 'threshold' if 0.49<= test_samples_sorted.loc[x,rep_gene_A]<=0.51
+                else 'greater',
+                samples
+            )      
+        )
+    )
+)
+data_encoded_labeled.head()
+
+
+# In[19]:
+
+
+# Heatmap sorted by gene expression signature
+# colormap: 
+#     offset - red
+#     less - blue
+#     threshold - green
+#     greater- magenta
+threshold_groups = data_encoded_labeled["threshold_group"]
+lut = dict(zip(threshold_groups.unique(), "rbgm"))
+row_colors = threshold_groups.map(lut)
+
+sns.clustermap(data_encoded,
```

Comment:
Can we change the cmap to viridis or something with a bit more cooler tone? On the other hand it looks like there isn't a change between samples that were assigned different groups. Is this a problem? Is this what you expect?